transformation/src/main/scala/org/broadinstitute/monster/hca/HcaPipelineBuilder.scala

```diff
@@ -85,20 +107,35 @@ object HcaPipelineBuilder extends PipelineBuilder[Args] {
   )
 
   /**
-    * Given a pattern matching JSON, get the JSONs as ReadableFiles.
+    * Given a pattern match, get the file MatchResult.Metadata.
     *
-    * @param jsonPath the root path containing JSONs to be converted
-    * @param context context of the main V2F pipeline
+    * @param filePattern the root path containing files to be matched
+    * @param context context of the main pipeline
     */
-  def getReadableFiles(jsonPath: String, context: ScioContext): SCollection[FileIO.ReadableFile] =
-    context.wrap {
+  def matchFiles(filePattern: String, context: ScioContext): SCollection[MatchResult.Metadata] = {
+    val metadata = context.wrap {
       context.pipeline.apply(
         FileIO
           .`match`()
-          .filepattern(jsonPath)
+          .filepattern(filePattern)
           .withEmptyMatchTreatment(EmptyMatchTreatment.ALLOW_IF_WILDCARD)
       )
-    }.applyTransform[ReadableFile](FileIO.readMatches())
+    }
+    metadata.count
+      .map(c => if (c == 0.toLong) NoMatchWarning(s"$filePattern had $c matches").log)
```

Comment:
I don't think `.toLong` is needed here.

Will this work? `map` returns a new `SCollection` but nothing reads it, so would it get forced to execute?

Reply:
It was executed when I was playing around with it 🙂